src/components/Cart/CartItems/CartItems.jsx

```diff
@@ -31,6 +41,47 @@ export const CartItems = () => {
     }
   };
 
+  const inWishlist = (_id) => {
+    const productInWishlist = wishlistState.wishlist.find(
+      (product) => product._id === _id
+    );
+
+    return productInWishlist ? true : false;
+  };
+
+  const listProducts = async () => {
+    try {
+      const response = await axios.get("/api/products");
+      if (response.status === 200) {
+        setProducts(response.data.products);
+      } else {
+        throw new Error();
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const addToWishlistHandler = async (_id) => {
+    const product = products.find((product) => product._id === _id);
+    if (token) {
+      if (!inWishlist(_id)) {
+        const response = await addToWishlistService(product, token);
+        if (response.status === 201) {
+          wishlistDispatch({
+            type: "ADD_TO_WISHLIST",
+            payload: response.data.wishlist,
+          });
+        }
```

Comment:
You could handle the error state in this case like other cases.

Reply:
Done